app/services/force/soql/preference_service.rb

```diff
@@ -34,15 +34,37 @@ def buildAppPreferencesFilters(opts)
 
         filters = ''
         filters += "and Preference_All_Name__c like '%#{opts[:preference]}' " if opts[:preference].present?
-        filters += "and Application__r.Name like '%#{opts[:application_number]}%' " if opts[:application_number].present?
-        filters += "and Application__r.Applicant__r.First_Name__c like '%#{opts[:first_name]}%' " if opts[:first_name].present?
-        filters += "and Application__r.Applicant__r.Last_Name__c like '%#{opts[:last_name]}%' " if opts[:last_name].present?
         filters += "and Application__r.Processing_Status__c = " + (opts[:status] == 'No Status' ? 'NULL' : "'#{opts[:status]}'") if opts[:status].present?
         filters += "and Application__r.Has_ADA_Priorities_Selected__c INCLUDES (#{accessibility_string}) " if opts[:accessibility].present?
 
         filters
     end
 
+    def buildAppPreferencesSearch(search_terms_string)
+      # Given a comma-separated list of search terms:
+      # For each term, we search across name and application number for a match
+      # For multiple terms, we expect the record to match each of the terms individually.
+      search_terms = search_terms_string.split(',')
+      search_fields = [
+        "Application__r.Name",
+        "Application__r.Applicant__r.First_Name__c",
+        "Application__r.Applicant__r.Last_Name__c",
+      ]
+      search = []
+      for search_term in search_terms
+        # For each term we join the fields with OR to return a record if it matches
+        # across any of these fields.
+        search_term_clause = []
+        sanitized_term = ActiveRecord::Base.sanitize_sql_like search_term
```

Comment:
tbh this doesn't seem to do much, but at least it escapes quotes and %'s so hopefully it helps